### ISSUE TITLE
Add rule: no-type-assertion-any

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#6]
## Problem

LLMs use `as any` as an escape hatch, bypassing TypeScript safety.

## Rule

Disallow type assertions to `any`.

### ❌ Bad
```ts
const result = data as any;
(item as any).property = value;
```

### ✅ Good
```ts
const result = data as unknown as ExpectedType;
```

## Why it matters

- `as any` is a code smell
- Bypasses all type safety
- Use `as unknown as T` if truly needed
